[2024-04-26T17:26:49.429+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-26T17:26:49.454+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [queued]>
[2024-04-26T17:26:49.460+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [queued]>
[2024-04-26T17:26:49.460+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-26T17:26:49.469+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): send_air_quality_alerts> on 2024-04-01 22:00:00+00:00
[2024-04-26T17:26:49.478+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=933) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-04-26T17:26:49.478+0000] {standard_task_runner.py:63} INFO - Started process 934 to run task
[2024-04-26T17:26:49.479+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'air_quality_pipeline', 'send_air_quality_alerts', 'backfill__2024-04-01T22:00:00+00:00', '--job-id', '359', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/air_quality/air_quality_dag.py', '--cfg-path', '/tmp/tmp6ectswfz']
[2024-04-26T17:26:49.480+0000] {standard_task_runner.py:91} INFO - Job 359: Subtask send_air_quality_alerts
[2024-04-26T17:26:49.512+0000] {task_command.py:426} INFO - Running <TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [running]> on host 4718ecf2379c
[2024-04-26T17:26:49.571+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='air_quality_pipeline' AIRFLOW_CTX_TASK_ID='send_air_quality_alerts' AIRFLOW_CTX_EXECUTION_DATE='2024-04-01T22:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='backfill__2024-04-01T22:00:00+00:00'
[2024-04-26T17:26:49.572+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-26T17:26:49.580+0000] {tasks.py:94} INFO - Generating air quality alerts...
[2024-04-26T17:26:49.582+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-26T17:26:49.582+0000] {taskinstance.py:2890} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 400, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 235, in execute
    return_value = self.execute_callable()
                   ^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/python.py", line 252, in execute_callable
    return self.python_callable(*self.op_args, **self.op_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/dags/air_quality/tasks.py", line 95, in send_air_quality_alerts
    alert_message = air_quality_alerts()
                    ^^^^^^^^^^^^^^^^^^^^
  File "/opt/airflow/src/air_quality_alerts.py", line 68, in air_quality_alerts
    data = pd.read_csv("air_quality_data.csv")
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 948, in read_csv
    return _read(filepath_or_buffer, kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 611, in _read
    parser = TextFileReader(filepath_or_buffer, **kwds)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1448, in __init__
    self._engine = self._make_engine(f, self.engine)
                   ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/parsers/readers.py", line 1705, in _make_engine
    self.handles = get_handle(
                   ^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/pandas/io/common.py", line 863, in get_handle
    handle = open(
             ^^^^^
FileNotFoundError: [Errno 2] No such file or directory: 'air_quality_data.csv'
[2024-04-26T17:26:49.590+0000] {taskinstance.py:1205} INFO - Marking task as UP_FOR_RETRY. dag_id=air_quality_pipeline, task_id=send_air_quality_alerts, execution_date=20240401T220000, start_date=20240426T172649, end_date=20240426T172649
[2024-04-26T17:26:49.597+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 359 for task send_air_quality_alerts ([Errno 2] No such file or directory: 'air_quality_data.csv'; 934)
[2024-04-26T17:26:49.611+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 1
[2024-04-26T17:26:49.626+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-26T17:26:49.627+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-04-26T17:31:41.673+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-26T17:31:41.697+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [queued]>
[2024-04-26T17:31:41.701+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [queued]>
[2024-04-26T17:31:41.702+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-26T17:31:41.709+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): send_air_quality_alerts> on 2024-04-01 22:00:00+00:00
[2024-04-26T17:31:41.717+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1239) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-04-26T17:31:41.718+0000] {standard_task_runner.py:63} INFO - Started process 1240 to run task
[2024-04-26T17:31:41.717+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'air_quality_pipeline', 'send_air_quality_alerts', 'backfill__2024-04-01T22:00:00+00:00', '--job-id', '375', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/air_quality/air_quality_dag.py', '--cfg-path', '/tmp/tmpio_q_xio']
[2024-04-26T17:31:41.718+0000] {standard_task_runner.py:91} INFO - Job 375: Subtask send_air_quality_alerts
[2024-04-26T17:31:41.751+0000] {task_command.py:426} INFO - Running <TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [running]> on host 4718ecf2379c
[2024-04-26T17:31:41.811+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='air_quality_pipeline' AIRFLOW_CTX_TASK_ID='send_air_quality_alerts' AIRFLOW_CTX_EXECUTION_DATE='2024-04-01T22:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='backfill__2024-04-01T22:00:00+00:00'
[2024-04-26T17:31:41.811+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-26T17:31:41.820+0000] {tasks.py:94} INFO - Generating air quality alerts...
[2024-04-26T17:31:41.827+0000] {tasks.py:102} INFO - Cleaning up temporary files...
[2024-04-26T17:31:41.828+0000] {tasks.py:105} INFO - Temporary files cleaned up.
[2024-04-26T17:31:41.828+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-04-26T17:31:41.828+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-26T17:31:41.835+0000] {taskinstance.py:1205} INFO - Marking task as SUCCESS. dag_id=air_quality_pipeline, task_id=send_air_quality_alerts, execution_date=20240401T220000, start_date=20240426T173141, end_date=20240426T173141
[2024-04-26T17:31:41.851+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-04-26T17:31:41.864+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-26T17:31:41.866+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
[2024-04-26T17:40:02.956+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2024-04-26T17:40:02.984+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [queued]>
[2024-04-26T17:40:02.989+0000] {taskinstance.py:2073} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [queued]>
[2024-04-26T17:40:02.989+0000] {taskinstance.py:2303} INFO - Starting attempt 1 of 2
[2024-04-26T17:40:02.998+0000] {taskinstance.py:2327} INFO - Executing <Task(PythonOperator): send_air_quality_alerts> on 2024-04-01 22:00:00+00:00
[2024-04-26T17:40:03.006+0000] {logging_mixin.py:188} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:61 DeprecationWarning: This process (pid=1573) is multi-threaded, use of fork() may lead to deadlocks in the child.
[2024-04-26T17:40:03.007+0000] {standard_task_runner.py:63} INFO - Started process 1574 to run task
[2024-04-26T17:40:03.006+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'air_quality_pipeline', 'send_air_quality_alerts', 'backfill__2024-04-01T22:00:00+00:00', '--job-id', '392', '--pool', 'default_pool', '--raw', '--subdir', 'DAGS_FOLDER/air_quality/air_quality_dag.py', '--cfg-path', '/tmp/tmpsbiv2m7l']
[2024-04-26T17:40:03.008+0000] {standard_task_runner.py:91} INFO - Job 392: Subtask send_air_quality_alerts
[2024-04-26T17:40:03.040+0000] {task_command.py:426} INFO - Running <TaskInstance: air_quality_pipeline.send_air_quality_alerts backfill__2024-04-01T22:00:00+00:00 [running]> on host 4718ecf2379c
[2024-04-26T17:40:03.102+0000] {taskinstance.py:2644} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='air_quality_pipeline' AIRFLOW_CTX_TASK_ID='send_air_quality_alerts' AIRFLOW_CTX_EXECUTION_DATE='2024-04-01T22:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='1' AIRFLOW_CTX_DAG_RUN_ID='backfill__2024-04-01T22:00:00+00:00'
[2024-04-26T17:40:03.103+0000] {taskinstance.py:430} INFO - ::endgroup::
[2024-04-26T17:40:03.113+0000] {tasks.py:94} INFO - Generating air quality alerts...
[2024-04-26T17:40:03.122+0000] {tasks.py:102} INFO - Cleaning up temporary files...
[2024-04-26T17:40:03.123+0000] {tasks.py:105} INFO - Temporary files cleaned up.
[2024-04-26T17:40:03.123+0000] {python.py:237} INFO - Done. Returned value was: None
[2024-04-26T17:40:03.123+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2024-04-26T17:40:03.132+0000] {taskinstance.py:1205} INFO - Marking task as SUCCESS. dag_id=air_quality_pipeline, task_id=send_air_quality_alerts, execution_date=20240401T220000, start_date=20240426T174002, end_date=20240426T174003
[2024-04-26T17:40:03.181+0000] {local_task_job_runner.py:240} INFO - Task exited with return code 0
[2024-04-26T17:40:03.195+0000] {taskinstance.py:3482} INFO - 0 downstream tasks scheduled from follow-on schedule check
[2024-04-26T17:40:03.196+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
